auditable-serde/src/lib.rs

```diff
@@ -160,28 +182,111 @@ impl From<&str> for Source {
 }
 
 impl From<Source> for String {
+    /// Provides a lossy conversion for variants with values different than the default
     fn from(s: Source) -> String {
         match s {
             Source::CratesIo => "crates.io".to_owned(),
-            Source::Git => "git".to_owned(),
+            Source::Git(_) => "git".to_owned(),
             Source::Local => "local".to_owned(),
             Source::Registry => "registry".to_owned(),
             Source::Other(string) => string,
         }
     }
 }
```

Comment:
This implementation along with `From<&str> for Source` make me feel the most conflicted. They are a part of publicly exposed API so changing their behavior could become problematic. For some reason I feel like I shouldn't just use serde_json and parse/serialize the value for git variant but I am not sure if the feeling is justified.